src/generators.h

```diff
@@ -66,6 +66,7 @@ struct GeneratorParams : std::enable_shared_from_this<GeneratorParams>, LeakChec
   int batch_size{1};
   int max_batch_size{0};
   bool use_cuda_graph{};
+  mutable bool session_terminated{false};
```

Comment:
```suggestion
  mutable bool session_terminated{};
```

Reply:
I like the initialization to be explicit, any harm in adding the false flag?

Reply:
Empty braces are zero/false initialization. We use this everywhere, even in the above code, max_batch_size{0} should also use it. Less to read.